### PR TITLE
[Lib + Test Fix] Fix nfs tests and add nfs utils

### PR DESCRIPTION
**Pull request by pranavprakash20**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
run results will be attached once the rhos-d issue is resolved.

**Review by MohitBis — DISMISSED** [review]
LGTM

Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-RG5WQT/

**Review by msainiRedhat — DISMISSED** [review]
LGTM

File: cli/utilities/nfs_utils.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    if not isinstance(clients, list):
+        clients = [clients]
+
+    for client in clients:
+        # Create mount dirs
+        client.exec_command(cmd=f"mkdir -p {nfs_mount}", sudo=True)
+        # Perform mount
+        cmd = f"mount -t nfs -o vers={version},port={port} {nfs_server_name}:{nfs_export} {nfs_mount}"
+        out, _ = client.exec_command(cmd=cmd, sudo=True)
+        if out:
```
[the remark this answers is not in the document]
Comment (pranavprakash20) [reply cli/utilities/nfs_utils.py:62]: yes, so if the mount fails, we need to exit right? out will be 1, is the mount comd fails, so if out: then return false and exit the process.
Am I missing something?

[the remark this answers is not in the document]
Comment (pranavprakash20) [reply cli/utilities/nfs_utils.py:62]: My mistake, got it. Will move return True outside loop

**Review by MohitBis — APPROVED** [review]
LGTM

**Review by msainiRedhat — APPROVED** [review]
LGTM

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Review by MohitBis — APPROVED** [review]
LGTM

Working as expected with below as well
https://github.com/red-hat-storage/cephci/pull/2745
log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-CQMR3C/

**Review by msainiRedhat — APPROVED** [review]
LGTM

**Review by vamahaja — APPROVED** [review]

**Pull request merged**